ReactWindows/ReactNative.Net46/Views/TextInput/PlaceholderAdorner.cs

```diff
@@ -0,0 +1,1104 @@
+        /// </summary>
+        /// <param name="adornedElement">
+        ///   The element to bind the adorner to.
+        /// </param>
+        /// <exception cref="T:System.ArgumentNullException">
+        ///   Raised when adornedElement is null.
+        /// </exception>
+        public PlaceholderAdorner(PasswordBox adornedElement)
+            : this((Control)adornedElement)
+        {
+            if (!(adornedElement.IsFocused && (bool)adornedElement.GetValue(HideOnFocusProperty)))
+            {
+                adornedElement.PasswordChanged += this.AdornedElement_ContentChanged;
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="T:T:ReactNative.Views.TextInput.PlaceholderAdorner"/> class.
+        /// </summary>
+        /// <param name="adornedElement">
+        ///   The element to bind the adorner to.
+        /// </param>
+        /// <exception cref="T:System.ArgumentNullException">
+        ///   Raised when adornedElement is null.
+        /// </exception>
+        public PlaceholderAdorner(TextBoxBase adornedElement)
+            : this((Control)adornedElement)
+        {
+
+            //if ((bool) adornedElement.GetValue(ShowWhenContentEmptyProperty))
+            //{
+            //    adornedElement.TextChanged += this.AdornedElement_ContentChanged;
+            //}
+
+            if (!(adornedElement.IsFocused && (bool)adornedElement.GetValue(HideOnFocusProperty)))
+            {
+                adornedElement.TextChanged += this.AdornedElement_ContentChanged;
+            }
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="T:T:ReactNative.Views.TextInput.PlaceholderAdorner"/> class.
+        /// </summary>
+        /// <param name="adornedElement">
+        ///   The element to bind the adorner to.
+        /// </param>
+        /// <exception cref="T:System.ArgumentNullException">
+        ///   Raised when adornedElement is null.
+        /// </exception>
+        protected PlaceholderAdorner(Control adornedElement)
+            : base(adornedElement)
+        {
+            if ((bool)adornedElement.GetValue(HideOnFocusProperty))
+            {
+                adornedElement.GotFocus += this.AdornedElement_GotFocus;
+                adornedElement.LostFocus += this.AdornedElement_LostFocus;
+            }
+        }
+        #endregion
+
+        #region Override Methods
+
+        /// <summary>
+        ///   Draws the content of a <see cref="T:System.Windows.Media.DrawingContext" /> object during the render pass of a <see cref="T:T:ReactNative.Views.TextInput.PlaceholderAdorner"/> element.
+        /// </summary>
+        /// <param name="drawingContext">
+        ///   The <see cref="T:System.Windows.Media.DrawingContext" /> object to draw. This context is provided to the layout system.
+        /// </param>
+        protected override void OnRender(DrawingContext drawingContext)
```

Comment:
The OnRender will draw the placeholder into the adorner layer... This method coalesces all the attached properties and draws the appropriately styled text.